src/utils/beacon/decodePayload.ts

```diff
@@ -0,0 +1,52 @@
+import { hex2buf } from "@taquito/utils";
+
+const PAYLOAD_PADDING = "0501";
+
+/**
+ * Decodes a sign request payload string.
+ *
+ * @param payload - The payload to decode.
+ * @returns The decoded string, or the original payload if decoding fails.
+ * @see {@link https://taquito.io/docs/signing/} for more info.
+ */
+export const decodePayload = (payload: string): string => {
+  let index = 0;
+
+  if (payload.startsWith(PAYLOAD_PADDING)) {
+    index = 4;
+
+    while (index < payload.length && payload[index] === "0") {
+      index += 1;
+    }
+
+    let rawBytesLength = "";
+    let bytesLength = 0;
+    const paddingLength = index;
+
+    while (index < payload.length && payload[index] !== "0") {
+      rawBytesLength += payload[index];
+      bytesLength = parseInt(rawBytesLength, 16);
+
+      const remainingPayloadLength = payload.length - rawBytesLength.length - paddingLength;
+
+      if (bytesLength * 2 === remainingPayloadLength) {
+        index += 1;
+        break;
+      }
+
+      if (bytesLength * 2 > remainingPayloadLength) {
+        return payload;
+      }
+
+      index += 1;
+    }
+  }
+
+  try {
+    const string = new TextDecoder("utf-8").decode(hex2buf(payload.slice(index)));
```

Comment:
[nit]

IIUC lines 13-43 are basically searching for the payload start index.

To improve readability maybe we can move them to a separate private method called something like `getPayloadHexBytes`, which
- returns payload.slice(index) to go into `TextDecoder("utf-8").decode(hex2buf())`
- throw error  or return null / empty string in case of incorrect length